grails-app/services/com/redhat/theses/listeners/RoleRequestListenerService.groovy

```diff
@@ -0,0 +1,75 @@
+package com.redhat.theses.listeners
+
+import com.redhat.theses.RoleRequest
+import com.redhat.theses.Feed
+import com.redhat.theses.events.RoleRequestEvent
+import com.redhat.theses.util.Util
+import com.redhat.theses.auth.User
+import com.redhat.theses.auth.Role
+import grails.events.Listener
+import org.springframework.context.i18n.LocaleContextHolder as LCH
+
+class RoleRequestListenerService {
+
+    def messageSource
+    /**
+     * Dependency injection of com.redhat.theses.SubscriptionService
+     */
+    def subscriptionService
+
+    /**
+     * Dependency injection of org.codehaus.groovy.grails.web.mapping.LinkGenerator
+     */
+    def grailsLinkGenerator
+
+    /**
+     * Creates new feed and notifies owner of applied topic about creation of new application
+     */
+    @Listener(topic = "RoleRequestCreated")
+    def RoleRequestCreated(RoleRequestEvent e) {
+        def request = e.request
+        def args = [
+                e.user.fullName,
+                grailsLinkGenerator.link(controller: 'user', action: 'show', id: e.user.id, absolute: true),
+                grailsLinkGenerator.link(controller: 'roleRequest', action: 'show', id: request.id, absolute: true)
+        ]
+
+        def feed = new Feed(messageCode: 'feed.request.created', args: args, user: e.user).save()
+
+        for (User admin : User.list().findAll { user -> Role.ADMIN in user.roles }) {
```

Comment:
Sadly, Grails doesn't generate these functions for enums, but I found better way to do it: Generate my own query to filter only Admins.